plugin/reload/README.md

```diff
@@ -59,3 +60,27 @@ Check every 10 seconds (jitter is automatically set to 10 / 2 = 5 in this case):
     erratic
 }
 ~~~
+
+## Bugs
+
+The reload happens without data loss (i.e. DNS queries keep flowing), but there is a corner case
+where the reload "succeeds", but you can still loose functionally. Consider the following Corefile:
+
+~~~ corefile
+. {
+	health :8080
+	whoami
+}
+~~~
+
+CoreDNS starts and serves health from :8080. Now you change `:8080` to `:443` not knowing a process
+is already listening on that port. The process reloads, and performs the following steps:
+
+1. close the listener on 8080
+2. reload and parse the config again
+3. fail to start a new listener on 443
+
+Now failing step 3 *does not cause* the reload to fail, instead almost everything works, except the
```

Comment:
Are you sure ?
I think that if the Listener fails to start on 443, then OnStartup will return an error, that will be returned back to Caddy .. and the Start of the new server will just fail, leading to the fail of Restart().

The old server will keep running, but the Listener will be closed on :8080

Reply:
changed